src/parser/internal/arrays.rs

```diff
@@ -15,68 +15,66 @@ pub fn list_expression(state: &mut State) -> ParseResult<Expression> {
     let mut items = Vec::new();
     let mut has_atleast_one_key = false;
 
-    while state.current.kind != TokenKind::RightParen {
-        if state.current.kind == TokenKind::Comma {
+    while state.tokens.current().kind != TokenKind::RightParen {
+        if state.tokens.current().kind == TokenKind::Comma {
             items.push(ListItem {
                 key: None,
                 value: Expression::Empty,
             });
 
-            state.next();
-            state.skip_comments();
+            state.tokens.next();
```

Comment:
all skip comments calls are removed, `next()`, `peek()`, and `lookahead()` are comment aware.